youtube_dl/extractor/bbc.py

```diff
@@ -959,6 +973,54 @@ def _real_extract(self, url):
              r'videoId\s*:\s*["\'](%s)["\']' % self._ID_REGEX],
             webpage, 'vpid', default=None)
 
+        # bbc reel (e.g. https://www.bbc.com/reel/video/p07c6sb6/how-positive-thinking-is-harming-your-happiness)
+        if not programme_id:
+            initial_data = self._html_search_regex(
+                r'<[sS][cC][rR][iI][pP][tT]\b[^>]+\bid\s*=\s*(["\'])initial-data\1[^>]+\bdata-json\s*=\s*(["\'])([^\']+)\2[^>]*>',
+                webpage, 'initial data', default=None, group=3)
+            if initial_data:
+                # let's see if it actually is, or warning
+                initial_data = self._parse_json(initial_data, 'initial data', fatal=False)
+                if initial_data:
+                    initial_data = try_get(initial_data,
+                                           lambda x: x['initData']['items'][0], dict)
+                    if initial_data:
+                        smp_data = initial_data.get('smpData', {})
+                        clip_data = try_get(smp_data, lambda x: x['items'][0], dict)
+                        if clip_data:
+                            initial_data.update(smp_data)
+                            initial_data.update(clip_data)
+                            programme_id = initial_data.get('versionID')
+            if programme_id:
+                alt_title = initial_data.get('shortTitle', playlist_id)
+                title = initial_data.get('title', alt_title)
+                if title == alt_title:
+                    alt_title = None
+                description = dict_get(initial_data, ['summary', 'shortSummary'])
+                timestamp = unified_timestamp(initial_data.get('displayDate'))
+                duration = initial_data.get('duration')
+                thumbnail = initial_data.get('holdingImageURL')
+                if thumbnail:
+                    thumbnail = thumbnail.replace('$recipe', '64x64')
+                    thumbnails = [
+                        {'url': thumbnail.replace('$recipe', '%dx%d' % (x, x)),
+                         'width': x, 'height': x}
+                        for x in [16, 32, 64, 128, 512]
+                    ]
+                else:
+                    thumbnails = None
+                formats, subtitles = self._download_media_selector(programme_id)
+                if formats:
+                    self._sort_formats(formats)
+                    vars = {'title', 'description', 'timestamp', 'duration', 'alt_title',
+                            'thumbnail', 'thumbnails', 'formats', 'subtitles'}
+                    ret = {'id': programme_id}
+                    ret.update([[x, locals().get(x)] for x in vars if locals().get(x)])
+                    return ret
+                else:
+                    self.report_warning('BBC Reel: no formats available', programme_id)
+                    programme_id = None
```

Comment:
not finding formats should and error, and it should be raised directly from `_sort_formats`.

Reply:
And it is, I see. You're saying that once we've got this far the extraction is committed and it doesn't make sense to trap the error as a warning and give further extraction techniques a chance?

I'll resolve this together with your next comment.